samples/AnomalyDetector/src/Generated/Models/MultivariateDetectionResult.Serialization.cs

```diff
@@ -33,12 +33,26 @@ void IJsonModel<MultivariateDetectionResult>.Write(Utf8JsonWriter writer, ModelR
                 writer.WriteStringValue(ResultId);
             }
             writer.WritePropertyName("summary"u8);
-            writer.WriteObjectValue(Summary);
+            if (Summary != null)
+            {
+                ((IJsonModel<MultivariateBatchDetectionResultSummary>)Summary).Write(writer, options);
+            }
+            else
+            {
+                writer.WriteNullValue();
+            }
```

Comment:
this is the third kind of the changes: required property.
Despite when constructing the model using public ctor, this property might not be allowed to be null, we have other ways to construct a model, in which almost any value is allowed for a property as long as csharp language allows, for example, the model factory. Using the model factory, you could construct an instance of a model with all of its properties to be null, no matter it is required or optional.
The null check here is required otherwise if we have an instance constructed by model factory, this serialization will throw NRE.